lib/shoryuken/queue.rb

```diff
@@ -37,38 +37,68 @@ def send_messages(options)
 
     def receive_messages(options)
       client.receive_message(options.merge(queue_url: url)).
-        messages.
-        map { |m| Message.new(client, self, m) }
+          messages.
+          map { |m| Message.new(client, self, m) }
+    end
+
+    # Returns whether this queue is a FIFO queue or not.
+    # @return [TrueClass, FalseClass]
+    def is_fifo?
+      fifo_attributes.attributes[FIFO_ATTRIBUTE] == 'true'
+    end
+
+    # Returns whether this queue has content based deduplication enabled or not.
+    # @return [TrueClass, FalseClass]
+    def has_content_deduplication?
+      fifo_attributes.attributes[CONTENT_DEDUP_ATTRIBUTE] == 'true'
     end
 
     private
 
+    FIFO_ATTRIBUTE          = 'FifoQueue'
+    CONTENT_DEDUP_ATTRIBUTE = 'ContentBasedDeduplication'
+    MESSAGE_GROUP_ID        = 'ShoryukenMessage'
+
+    # @return [Aws::SQS::Types::GetQueueAttributesResult]
+    def fifo_attributes
+      @fifo_attr ||= client.get_queue_attributes(queue_url: url, attribute_names: [FIFO_ATTRIBUTE, CONTENT_DEDUP_ATTRIBUTE])
```

Comment:
I was wondering if we could do:

```ruby
def attributes
  @_attributes || client.get_queue_attributes(queue_url: url, attribute_names: [FIFO_ATTRIBUTE, CONTENT_DEDUP_ATTRIBUTE, 'VisibilityTimeout'])
end
```

So we could share the `attributes` with `visibility_timeout` and `fifo`.